src/secure_cookie/cookie.py

```diff
@@ -309,6 +325,48 @@ def unserialize(cls, string, secret_key):
         if isinstance(secret_key, text_type):
             secret_key = secret_key.encode("utf-8", "replace")
 
+        signer = Signer(secret_key, digest_method=cls.hash_method)
+        try:
+            serialized = signer.unsign(string)
+        except BadSignature:
+            return cls._mac_unserialize(string, secret_key)
+
+        items = {}
+        for item in serialized.split(b"&"):
+            if b"=" not in item:
+                items = None
+                break
+
+            key, value = item.split(b"=", 1)
+            # try to make the key a string
+            key = url_unquote_plus(key.decode("ascii"))
+
+            try:
+                key = to_native(key)
+            except UnicodeError:
+                pass
+
+            items[key] = value
+
+        if items is not None:
+            try:
+                for key, value in items.items():
+                    items[key] = cls.unquote(value)
+            except UnquoteError:
+                items = ()
+            else:
+                if "_expires" in items:
+                    if time() > items["_expires"]:
+                        items = ()
+                    else:
+                        del items["_expires"]
+        else:
+            items = ()
+        return cls(items, secret_key, False)
+
+    @classmethod
+    def _mac_unserialize(cls, string, secret_key):
+        warnings.warn("Obsolete serialization method used", DeprecationWarning)
```

Comment:
I updated this part. The new string is longer than required by black so I added a `# noqa` comment to ignore it. If you prefer, I can extract it in a variable to reformat it.
I don't think it's necessary because the `_mac_unserialize()` method will disappear in a future release. What is your opinion about it?